service/share/share.go

```diff
@@ -22,21 +21,15 @@ import (
 
 var log = logging.Logger("share")
 
-// DefaultCodec sets the default rsmt2d.Codec for shares.
-var DefaultCodec = consts.DefaultCodec
-
-// TODO(@Wondertan): We prefix real data of shares with namespaces to be able to recover them during erasure coding
-//  recovery. However, that is storage and bandwidth overhead(8 bytes per each share) which we can avoid by getting
-//  namespaces from CIDs stored in IPLD NMT Nodes, instead of encoding namespaces in erasure coding.
-//
-// TODO(@Wondertan): Ideally, Shares structure should be defined in a separate store with
-//  rsmt2d/nmt/celestia-core/celestia-node importing it. rsmt2d and nmt are already dependent on the notion of "share",
-//  so why shouldn't we have a separated and global type for it to avoid the type mess with defining own share type in
-//  each package.
-//
 // Share is a fixed-size data chunk associated with a namespace ID, whose data will be erasure-coded and committed
 // to in Namespace Merkle trees.
-type Share = namespace.PrefixedData8
+type Share = ipld.Share
+
+// GetID extracts namespace ID out of a Share.
+var GetID = ipld.ShareID
+
+// GetData extracts data out of a Share.
+var GetData = ipld.ShareData
```

Comment:
I took me a while to notice that these are functions 🙈 
It seems like GetData is not (yet?) used anywhere.

Reply:
Yes. It is not used yet. Unfortunately, I had to define those as funcs and not as methods, because its not possible to define methods for byte aliases in Go.

Reply:
Now its used after rebase on the main